spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/source/TestIcebergSourceTablesBase.java

```diff
@@ -1382,6 +1404,76 @@ public void testPartitionsTable() {
       TestHelpers.assertEqualsSafe(
           partitionsTable.schema().asStruct(), expected.get(i), actual.get(i));
     }
+
+    testDeleteStats(tableIdentifier, table, partitionsTable, builder, partitionBuilder, expected);
```

Comment:
Could we make this test independently?  As the existing test is already quite long, and it's doing more table modifications inside the new method.

Reply:
done.